ios/brave-ios/Sources/Brave/Frontend/Sync/BraveCore/History/HistoryAPIExtensions.swift

```diff
@@ -25,29 +25,31 @@ extension BraveHistoryAPI {
       maxCount: UInt(max(20, maxLength)),
       duplicateHandling: .removePerDay
     )
+
     search(
       withQuery: nil,
       options: options,
-      completion: { historyResults in
-        completion(historyResults.map { $0 })
-      }
+      completion: completion
     )
   }
 
-  func byFrequency(query: String, completion: @escaping ([HistoryNode]) -> Void) {
-    guard !query.isEmpty else {
-      return
+  func byFrequency(
+    query: String,
+    completion: @escaping ([HistoryNode]) -> Void
+  ) -> HistoryCancellable? {
+    if query.isEmpty {
+      return nil
     }
+
     let options = HistorySearchOptions(
       maxCount: 200,
-      duplicateHandling: query.isEmpty ? .removePerDay : .removeAll
+      duplicateHandling: .removeAll
     )
```

Comment:
why we change the duplicateHandling option?

Reply:
Previously the code was: 
<img width="366" alt="image" src="https://github.com/user-attachments/assets/5894367c-bdf1-4b3e-9cd4-9fb66035c390" />

It would never enter the function is `query.isEmpty`, it just returns `nil`. So I think either the code is wrong, OR the duplicate handling is wrong.